src/components/invite/useInvites.js

```diff
@@ -0,0 +1,131 @@
+import { useEffect, useState } from 'react'
+import { keyBy, uniqBy } from 'lodash'
+import goodWallet from '../../lib/wallet/GoodWallet'
+import userStorage from '../../lib/gundb/UserStorage'
+import logger from '../../lib/logger/pino-logger'
+import AsyncStorage from '../../lib/utils/asyncStorage'
+import { useDialog } from '../../lib/undux/utils/dialog'
+import { fireEvent, INVITE_BOUNTY, INVITE_JOIN } from '../../lib/analytics/analytics'
+
+const log = logger.child({ from: 'useInvites' })
+
+const registerForInvites = async () => {
+  const inviterCode = userStorage.userProperties.get('inviterCode')
+  if (inviterCode) {
+    fireEvent(INVITE_JOIN, { inviterCode })
+  }
+  const inviteCode = await goodWallet.joinInvites(inviterCode)
+  log.debug('joined invites contract:', { inviteCode, inviterCode })
+  userStorage.userProperties.set('inviteCode', inviteCode)
+  return inviteCode
+}
+
+export const getInviteCode = async () => {
```

Comment:
```suggestion
const getInviteCode = async () => {
```

Reply:
better to use the hook instead of 'private' functions in AppSwitch, so export doesn't needed